Classes/Authentication/AuthService.php

```diff
@@ -45,9 +48,14 @@ public function getUser(): array|bool
     public function authUser(array $user): int
     {
         $result = 100;
-        if (VerificationUtility::verifyImpersonateData(GeneralUtility::_GET('tx_impersonate'))) {
+        if (VerificationUtility::verifyImpersonateData($this->getRequest()->getQueryParams()['tx_impersonate'])) {
             $result = 200;
         }
         return $result;
     }
+
+    private function getRequest(): ServerRequestInterface
+    {
+        return ServerRequestFactory::fromGlobals();
```

Comment:
can here also `$GLOBALS['TYPO3_REQUEST']` be used?

Reply:
Nope, not available in this context. Tried that before.